### PR TITLE
Updates the trafficcontrol-health-client with the following.

### PR DESCRIPTION
**Pull request by jrushford**
Updates the trafficcontrol-health-client with the following:

- Add the "unavailable-poll-threshold" configuration parameter.
  A parent must be polled as unhealthy for a consecutive number
  of times.  When the unhealthy poll count reaches this threshold
  the parent is marked down.
- Fixes a bug where a change to the "tm-poll-interval-seconds" is
  not applied to a running health-client when it re-reads it's
  configuration file.

## Which Traffic Control components are affected by this PR?

- Traffic Control Health Client (tc-health-client)

## What is the best way to verify this PR?

Ensure unit tests pass and when running that an unhealthy parent is not marked down until the 
number of unhealthy polls for that parent meets or exceeds the "unavailable-poll-threshold"
which defaults to 2.

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by jpappa200 — APPROVED** [review]
👍

**Review by ocket8888 — APPROVED** [review]

File: tc-health-client/tmagent/tmagent.go [hunk 1]
```diff
@@ -472,27 +476,81 @@ func (c *ParentInfo) markParent(fqdn string, available bool) error {
...
 	}
+
+	return nil
+}
+
+// used to mark a parent as up or down in the trafficserver HostStatus
+// subsystem.
+//
+// TODO see issue #6448, add cacheStatus back when available in CrStates
+//func (c *ParentInfo) markParent(fqdn string, cacheStatus string, available bool) error {
```
Comment (zrhoffman) [inline tc-health-client/tmagent/tmagent.go:487]: This commented line should be removed.

File: tc-health-client/tmagent/tmagent.go [hunk 2]
```diff
@@ -472,27 +476,81 @@ func (c *ParentInfo) markParent(fqdn string, available bool) error {
...
+				hostAvailable = true
+			} else {
+				// marking the host down
+				err = c.execTrafficCtl(fqdn, available)
+				if err != nil {
+					log.Errorln(err.Error())
+				}
+				if err == nil {
+					// TODO see issue 6448, add cacheStatus back when available in CrStates
+					// log.Infof("marked parent %s DOWN, cache status was: %s\n", hostName, cacheStatus)
```
Comment (zrhoffman) [inline tc-health-client/tmagent/tmagent.go:516]: This commented line should be removed.

File: tc-health-client/tmagent/tmagent.go [hunk 3]
```diff
@@ -472,27 +476,81 @@ func (c *ParentInfo) markParent(fqdn string, available bool) error {
...
+			}
+		} else { // available
+			// marking the host up
+			err = c.execTrafficCtl(fqdn, available)
+			if err == nil {
+				hostAvailable = true
+				// reset the unavilable poll count
+				unavailablePollCount = 0
+				// TODO see issue #6448, add cacheStatus back when available in CrStates
+				//log.Infof("marked parent %s UP, cache status was: %s\n", hostName, cacheStatus)
```
Comment (zrhoffman) [inline tc-health-client/tmagent/tmagent.go:529]: This commented line should be removed.

**Comment by zrhoffman** [comment]
Just noticed the *TODO* comments, now those lines make sense.

**Pull request merged**